### DISCUSSION TITLE
Poe by Quora

### DISCUSSION THREAD

**Discussion by omodia** [mienaiyami/yomikiru#237] [General]
1. You can chat with [ChatGPT using this website](https://poe.com/ChatGPT). No phone number verification. Only email verification.
- Chat limit - 10 questions per 24 hours for free accounts.
- It cannot parse long lines of code directly using chat box.
- Instead upload text files. Can parse complete large text files. (Only .txt format, and not .tsx format).
- Upload one text file per one message. It parses individual text files better than uploading them in batch.

2. For example, I had uploaded a couple of text files from components after saving them as .txt format and asked it to identify performance improvements and potential bugs. I don't know if its helpful or not. But, I uploaded the reply as a text file.

- [ChatGPT - Reader.txt](https://github.com/mienaiyami/yomikiru/files/12310672/ChatGPT.-.Reader.txt)
- [ChatGPT.txt](https://github.com/mienaiyami/yomikiru/files/12310671/ChatGPT.txt) - I don't remember which file I had uploaded, hence there is no name for this text file.

**Comment by omodia** [comment]
Note that it does not have knowledge of events that took place after 2021.

**Comment by omodia** [comment]
I had uploaded all the .tsx files in components and the bot used is ChatGPT 3.5.
Extract the zip and open the .mhtml file in chrome to see the reply given by the bot.

### [ChatGPT.zip](https://github.com/mienaiyami/yomikiru/files/12314109/ChatGPT.zip)

**Comment by mienaiyami** [comment]
Thank you very much for all this. I ensure the use of proper linting and extensions to optimize code quality and performance. 
I'm aware of all the points mentioned in these .txt, and I have valid reasons for implementing them, which ChatGPT won't recognize. I specifically made some functions to not be stored in RAM and use CPU at runtime, and some to use RAM and not use CPU. However, I must admit that when I began developing this app, I was relatively inexperienced. The app still contains some outdated code that requires improvements, and I do my best to enhance them when possible, minimizing the time investment; otherwise, I mark them for later attention. 

I noted that it uses ChatGPT 3.5, and I think it is a bit lacking and not very accurate. While it was the reason for fixing the very first biggest issues (scroll stuttering), there weren't any right answers, I just got some idea from the solutions it provided. 

I never really expected this app to get this big and complex. You have been here since the very beginning so you know how I was not really contributing anything to it anymore. It's all thanks to you and a few others who created issues and motivated me to work more on the app and helped this app go forward. 

Once again, I express my gratitude for your time. Lately, I'm trying to fix up things as quickly as possible because I would like to work on more projects. While I genuinely enjoy working on this project, I need to create projects that can enhance my portfolio and provide a potential income source to support my college years and later.

**Reply by omodia** [reply]
You are a great developer, so you will land a job.
This app indeed came a long way and became the best reader to read manga, manhua, webtoon and epub's.
I thank you for the time you had invested and are still investing in this app.

Finally, thank you for the kind words.

**Comment by omodia** [comment]
You might know this already, but you can use edge to chat with chatgpt 4 for free.

**Reply by mienaiyami** [reply]
You mean bing chat? Its only good for web results in my opinion nothing more. And cant be personalized like normal chatgpt.

**Reply by omodia** [reply]
I had tried it using the historytab file. You could check if its response is helpful.

---
### GPT-4

After analyzing the code you shared, here are some suggestions for potential performance improvements and other considerations:

- **Performance improvements:**
  * The `filter` function for the `history` array is called on every render of the component, which could potentially impact performance if the `history` array is large. One way to improve performance would be to use the `useMemo` hook from React to memoize the filtered results and only recompute them when the `history` or `filter` state variables change.
  * The regular expression used in the filter function is created on every render of the component. This could be optimized by using the `useMemo` hook to only create a new regular expression when the value of the `filter` state variable changes.

- **Other considerations:**
  * The `onKeyDown` event handler for the search input includes logic for handling various key presses, including preventing certain characters from being entered into the search input. However, it's not clear why these characters are being prevented, and it may be worth considering whether this is necessary or if there is a better way to handle this.
  * The code includes several instances where TypeScript's non-null assertion operator (`!`) is used to assert that a value is not null or undefined. While this can be useful in some cases, it can also potentially lead to runtime errors if the value being asserted is actually null or undefined. It may be worth considering whether there are safer ways to handle these cases, such as using optional chaining (`?.`) or providing default values.

These are just some suggestions based on my analysis of the code you shared. I hope they are helpful! 😊

**Reply by mienaiyami** [reply]
Thanks.
But the results are not always useful, you see, it is trained on many data which include dev blogs, and most are just copy paste of other with minimal changes. And some of them are just bad. 
As in the comment above, it mentioned the use of `useMemo` and such, will be a burden for RAM, and definitely not a performance issue, from what I have noticed till now. 

While I do admit that there will be many mistakes in my code, some of them are essential, even if they look bad or unsafe from other POV.

**Reply by omodia** [reply]
> While I do admit that there will be many mistakes in my code, some of them are essential, even if they look bad or unsafe from other POV.

True. Thanks for explaining.
